editor/loadFile.js

```diff
@@ -0,0 +1,26 @@
+function createFileLoader (moonchild, channel) {
+  var fileLoader = Object.create(null);
+
+  // assuming that the editor doesn't get changed
+  // weird stuff will happen if the editor -does- get changed
+  var editor = moonchild.getEditor()._codeMirror;
+  var filePath;
+
+  channel.on("fileLoad", function (messageData) {
+    // assuming JSON structure of form
+    // {
+    //   ...
+    //   content: "",
+    //   filePath: "bla.js"
+    //   ...
+    // }
+
+    var text = messageData.content;
+    filePath = messageData.filePath;
+
+    editor.setValue(text);
+
+    // let the editor know that a change happened
+    moonchild.poke();
```

Comment:
Are you talking about the file loading mechanism as a whole? Because I've made it in such a way that it's loadable as a modular plugin. I don't see the need to move this functionality to editor.js, as that has always resulted in an ever-growing pile of code for my projects.